test/common/common/log_macros_test.cc

```diff
@@ -37,4 +37,65 @@ TEST(Logger, All) {
   // Misc logging with no facility.
   ENVOY_LOG_MISC(info, "fake message");
 }
+
+TEST(Logger, evaluateParams) {
+  uint32_t i = 1;
+
+  // set logger's level to low level.
+  // log message with higher severity and make sure that params were evaluated.
+  GET_MISC_LOGGER().set_level(spdlog::level::info);
+  ENVOY_LOG_MISC(warn, "test message '{}'", i++);
+  ASSERT_THAT(i, testing::Eq(2));
```

Comment:
nit: prefer `EXPECT*` for non-fatal failures (it won't cause a crash later in the test), since `ASSERT*` will essentially abort the test case.  Here and elsewhere.

Reply:
Good suggestion. Done.